src/basic/src/ScalarFunction.C

```diff
@@ -35,8 +38,27 @@ BaseScalarFunction<V, M>::BaseScalarFunction(const char * prefix,
     const VectorSet<V, M> & domainSet)
   : m_env(domainSet.env()),
     m_prefix((std::string)(prefix) + "func_"),
-    m_domainSet(domainSet)
+    m_domainSet(domainSet),
+#ifndef DISABLE_BOOST_PROGRAM_OPTIONS
+    m_parser(new BoostInputOptionsParser(m_env.optionsInputFileName())),
+#endif
+    m_fdStepSize(QUESO_BASESCALARFN_FD_STEPSIZE_ODV)
 {
+  // Snarf fd step size from input file.
+#ifndef DISABLE_BOOST_PROGRAM_OPTIONS
+  m_parser->registerOption<double>(m_prefix + "fdStepSize",
+                                   QUESO_BASESCALARFN_FD_STEPSIZE_ODV,
+                                   "step size for finite difference");
+  m_parser->scanInputFile();
+  m_parser->getOption<double>(m_prefix + "fdStepSize", m_fdStepSize);
+#else
+  m_fdStepSize = m_env.input()(m_prefix + "fdStepSize",
+                               QUESO_BASESCALARFN_FD_STEPSIZE_ODV);
+#endif
```

Comment:
This is a good start, but it isn't always going to be fine-grained enough, at least not for users who aren't normalizing all their parameters.  When trying to do UQ with both a reaction rate preexponential coefficient of ~1e11 and a nitridation rate of ~0.3, there's no way to come up with a single perturbation distance that works well for both.

Take a look at the numerical_jacobian_h* stuff in https://github.com/libMesh/libmesh/blob/master/include/systems/fem_system.h

We let users override a gross numerical_jacobian_h, but optionally then they can also (or instead) individually override numerical_jacobian_h_for_var on one (or two or all) of the system's variables.  This wasn't coded out of forethought, either, this was necessary to get proper quasi-Newton convergence rates on some nasty un-normalized physics.